test/src/test/java/org/corfudb/runtime/view/ManagementViewTest.java

```diff
@@ -1664,4 +1658,97 @@ public void unblockSequencerRecoveryOnDeadlock() throws Exception {
+        addServerRule(SERVERS.PORT_0, new TestRule().always().drop());
+
+        // Append data 10 - 19
+        for (int i = 0; i < num; i++) {
+            streamView.append(payload);
+        }
+
+        // Remove partition and allow PORT_0 to be healed.
+        clearServerRules(SERVERS.PORT_0);
+
+        // Add rules so that Full Sequencer bootstrap messages are not sent to PORT_0.
+        TestRule dropFullBootstrapRule = new TestRule()
+                .matches(corfuMsg -> corfuMsg.getMsgType().equals(CorfuMsgType.BOOTSTRAP_SEQUENCER)
+                        && !((CorfuPayloadMsg<SequencerTailsRecoveryMsg>) corfuMsg).getPayload()
+                        .getBootstrapWithoutTailsUpdate()).drop();
+        addClientRule(getManagementServer(SERVERS.PORT_0).getManagementAgent().getCorfuRuntime(),
+                SERVERS.ENDPOINT_0, dropFullBootstrapRule);
+        addClientRule(getManagementServer(SERVERS.PORT_1).getManagementAgent().getCorfuRuntime(),
+                SERVERS.ENDPOINT_0, dropFullBootstrapRule);
+        addClientRule(getManagementServer(SERVERS.PORT_2).getManagementAgent().getCorfuRuntime(),
+                SERVERS.ENDPOINT_0, dropFullBootstrapRule);
+
+        // Wait until PORT_0 is responsive and healed in the layout.
+        for (int i = 0; i < PARAMETERS.NUM_ITERATIONS_MODERATE; i++) {
+            if (corfuRuntime.getLayoutView().getLayout().getUnresponsiveServers().isEmpty()) {
+                break;
+            }
+            Sleep.sleepUninterruptibly(PARAMETERS.TIMEOUT_SHORT);
+            corfuRuntime.invalidateLayout();
+        }
+        assertThat(corfuRuntime.getLayoutView().getLayout().getUnresponsiveServers().isEmpty())
+                .isTrue();
+
+        // Add a rule to partition out PORT_1.
+        addServerRule(SERVERS.PORT_1, new TestRule().always().drop());
+
+        // Wait until PORT_1 is marked unresponsive.
+        for (int i = 0; i < PARAMETERS.NUM_ITERATIONS_MODERATE; i++) {
+            if (!corfuRuntime.getLayoutView().getLayout().getUnresponsiveServers().isEmpty()) {
+                break;
+            }
+            Sleep.sleepUninterruptibly(PARAMETERS.TIMEOUT_SHORT);
+            corfuRuntime.invalidateLayout();
+        }
+        assertThat(corfuRuntime.getLayoutView().getLayout().getUnresponsiveServers().isEmpty())
+                .isFalse();
+
+        // Remove partition and allow PORT_1 to be healed.
+        clearServerRules(SERVERS.PORT_1);
+
+        // Wait until PORT_1 is responsive and healed in the layout.
+        for (int i = 0; i < PARAMETERS.NUM_ITERATIONS_MODERATE; i++) {
+            if (corfuRuntime.getLayoutView().getLayout().getUnresponsiveServers().isEmpty()) {
+                break;
+            }
+            Sleep.sleepUninterruptibly(PARAMETERS.TIMEOUT_SHORT);
+            corfuRuntime.invalidateLayout();
+        }
+        assertThat(corfuRuntime.getLayoutView().getLayout().getUnresponsiveServers().isEmpty())
```

Comment:
Right after PORT_1 comes back and to verify functionality, is there a way we could proof that the sequencer received a delta bootstrap and it rejected it? before allowing full sync messages back?

Reply:
I can intercept the delta bootstrap message but cannot verify that the sequencer rejects it. However, this would change in the future if we move the state creation to the sequencer.

Reply:
If you don't consider this check adds any value then it's ok, we can leave it as it is...

Reply:
I have added the message interception and check that we receive a delta bootstrap message and is responded with a NACK. Let me know if this looks good and if you have any other suggestions.